pkg/mkimage-raw-efi/install

```diff
@@ -170,24 +171,57 @@ done
 # we may be asked to pause before install procedure
 grep -q eve_pause_before_install /proc/cmdline && pause "formatting the /dev/$INSTALL_DEV"
 
-# FIXME: gross hack to allow P3 partition on a disk different from a bootable one
-# this needs to go away the moment we implement LVM or ZFS
+DISKS_TO_MERGE=""
+DISK_WITH_P3=""
 if [ "$INSTALL_DEV" != "$INSTALL_PERSIST" ]; then
-   # apparently sgdisk -Z doesn't clear MBR and keeps complaining
-   PDEV="/dev/$INSTALL_PERSIST"
-   dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc
-   sgdisk -Z --clear "$PDEV" 2>/dev/null || :
-   sgdisk --new 1:2048:0 --typecode=1:5f24425a-2dfa-11e8-a270-7b663faccc2c --change-name=1:P3 "$PDEV"
-   sgdisk -v "$PDEV"
-   # force make-raw to skip persist
-   MAKE_RAW_PARTS="efi imga imgb conf"
+   if echo "$INSTALL_PERSIST"| grep -q ","; then
+     modprobe zfs
+     MAKE_RAW_PARTS="efi imga imgb conf"
+     IFS=',' ;for dev in $INSTALL_PERSIST; do
+        if [ "$dev" = "$INSTALL_DEV" ]; then
+           # in case we want to have one of P3 on the bootable disk
+           MAKE_RAW_PARTS="efi imga imgb conf persist"
+           DISK_WITH_P3="$dev"
+        else
+           PDEV="/dev/$dev"
+           dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc
+           sgdisk -Z --clear "$PDEV" 2>/dev/null || :
+           DISKS_TO_MERGE="$DISKS_TO_MERGE $PDEV"
+        fi
+     done
+   else
+     # apparently sgdisk -Z doesn't clear MBR and keeps complaining
+     PDEV="/dev/$INSTALL_PERSIST"
+     dd if=/dev/zero of="$PDEV" bs=512 count=1 conv=notrunc
+     sgdisk -Z --clear "$PDEV" 2>/dev/null || :
+     sgdisk --new 1:2048:0 --typecode=1:5f24425a-2dfa-11e8-a270-7b663faccc2c --change-name=1:P3 "$PDEV"
+     sgdisk -v "$PDEV"
+     # force make-raw to skip persist
+     MAKE_RAW_PARTS="efi imga imgb conf"
+   fi
 fi
 
 # do the install (unless we're only here to collect the black box)
 grep -q eve_blackbox /proc/cmdline || /make-raw "/dev/$INSTALL_DEV" $MAKE_RAW_PARTS || bail "Installation failed. Entering shell..."
 
-# now the disk is ready - mount partitions
-mount_part P3 "$INSTALL_DEV" 2>/dev/null
+if [ "$DISKS_TO_MERGE" != "" ]; then
+  if [ "$DISK_WITH_P3" != "" ]; then
+    P3_ID="$(find_part P3 "$DISK_WITH_P3")"
+    [ -z "$P3_ID" ] && bail "Installation failed. Cannot found P3. Entering shell..."
+    DISKS_TO_MERGE="/dev/$P3_ID $DISKS_TO_MERGE"
+  fi
+  [ -e /root/sys ] || mkdir /root/sys && mount -t sysfs sysfs /root/sys
+  [ -e /root/proc ] || mkdir /root/proc && mount -t proc proc /root/proc
+  [ -e /root/dev ] || mkdir /root/dev && mount -t devtmpfs -o size=10m,nr_inodes=248418,mode=755,nosuid,noexec,relatime devtmpfs /root/dev
+  CMD="chroot /root zpool create -f -m none -o feature@encryption=enabled -O overlay=on persist $DISKS_TO_MERGE"
```

Comment:
This is just striping across the disks, right? I don't think that is useful. raidz2 would be useful, and raidz if we have too few disks for raidz2. But we would also potentially want to figure out if we have a mixture of SSDs and rotating disks, the we potentially want to place the zil on the ssd(s).

So it isn't clear to me what use case you have in mind for this.

Reply:
Good point. My variant is the simplest option. 
Regarding to the raid I can think about options to add.

Reply:
What do you think, if I will allow to write something like `eve_persist_disk=raidz1,sda,sdb,sdc,log,mirror,sdd,sde,cache,sdf`?

Reply:
Now it support complex notations

Reply:
I'd actually prefer if we figure that out based on the number of disks listed and not have our users have to be experts on how to setup the ZFS pool.

Reply:
So if 2 we do mirror, 3 raidz1, 4 and larger raidz2.
Then we can evolve the default selection over time.

Reply:
Modified